src/mem/entropy.h

```diff
@@ -0,0 +1,93 @@
+#include <cstdint>
+#include <random>
+#include <type_traits>
+
+#include "../ds/address.h"
+
+namespace snmalloc
+{
+  template<typename Pal>
+  std::enable_if_t<pal_supports<Entropy, Pal>, uint64_t> get_entropy64()
+  {
+    return Pal::get_entropy64();
+  }
+
+  template<typename Pal>
+  std::enable_if_t<!pal_supports<Entropy, Pal>, uint64_t> get_entropy64()
+  {
+    std::random_device rd;
+    uint64_t a = rd();
+    return (a << 32) ^ rd();
+  }
+
+  class LocalEntropy
```

Comment:
Maybe split into a class for `bit_source` and `constant_key` and one for `local_key` and `local_counter`?  That'd make the use elsewhere in the tree a little more obvious, I think.  (e.g., `get_next` is used only to populate `constant_key` and `bit_source` in `init()` here.)

Reply:
I agree and disagree on this. 

`refresh_bits` calls `get_next` so uses most of it.  

I think `get_next` might be useful in other parts, as we could generate new keys for each free list.  Leaving this open makes the refactoring easier at the moment.